<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-проекта-и-цель" data-toc-modified-id="Описание-проекта-и-цель-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание проекта и цель</a></span></li><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Описание данных</a></span></li><li><span><a href="#Загрузка-библиотек" data-toc-modified-id="Загрузка-библиотек-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Загрузка библиотек</a></span></li><li><span><a href="#Число-государственных,-частных-и-участковых-(сельских)-больничных-учреждений.-Ресурсы-и-деятельность-организаций-здравоохранения." data-toc-modified-id="Число-государственных,-частных-и-участковых-(сельских)-больничных-учреждений.-Ресурсы-и-деятельность-организаций-здравоохранения.-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Число государственных, частных и участковых (сельских) больничных учреждений. Ресурсы и деятельность организаций здравоохранения.</a></span><ul class="toc-item"><li><span><a href="#Предположение" data-toc-modified-id="Предположение-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Предположение</a></span></li><li><span><a href="#Загрузка-и-предобработка-данных" data-toc-modified-id="Загрузка-и-предобработка-данных-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Загрузка и предобработка данных</a></span></li><li><span><a href="#Соединение-в-общий-датасет" data-toc-modified-id="Соединение-в-общий-датасет-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Соединение в общий датасет</a></span></li><li><span><a href="#Создание-датасетов" data-toc-modified-id="Создание-датасетов-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Создание датасетов</a></span><ul class="toc-item"><li><span><a href="#Датасет-с-общими-значениями" data-toc-modified-id="Датасет-с-общими-значениями-4.4.1"><span class="toc-item-num">4.4.1&nbsp;&nbsp;</span>Датасет с общими значениями</a></span></li><li><span><a href="#Датасет-городских-поселений" data-toc-modified-id="Датасет-городских-поселений-4.4.2"><span class="toc-item-num">4.4.2&nbsp;&nbsp;</span>Датасет городских поселений</a></span></li><li><span><a href="#Датасет-сельской-местности" data-toc-modified-id="Датасет-сельской-местности-4.4.3"><span class="toc-item-num">4.4.3&nbsp;&nbsp;</span>Датасет сельской местности</a></span></li></ul></li><li><span><a href="#Транспонирование-ранее-созданных-3-х-датасетов" data-toc-modified-id="Транспонирование-ранее-созданных-3-х-датасетов-4.5"><span class="toc-item-num">4.5&nbsp;&nbsp;</span>Транспонирование ранее созданных 3-х датасетов</a></span><ul class="toc-item"><li><span><a href="#Транспонированный-датасет-с-общими-значениями" data-toc-modified-id="Транспонированный-датасет-с-общими-значениями-4.5.1"><span class="toc-item-num">4.5.1&nbsp;&nbsp;</span>Транспонированный датасет с общими значениями</a></span></li><li><span><a href="#Транспонированный-датасет-с-городскими-поселениями" data-toc-modified-id="Транспонированный-датасет-с-городскими-поселениями-4.5.2"><span class="toc-item-num">4.5.2&nbsp;&nbsp;</span>Транспонированный датасет с городскими поселениями</a></span></li><li><span><a href="#Транспонированный-датасет-сельской-местности" data-toc-modified-id="Транспонированный-датасет-сельской-местности-4.5.3"><span class="toc-item-num">4.5.3&nbsp;&nbsp;</span>Транспонированный датасет сельской местности</a></span></li></ul></li><li><span><a href="#Сохранение-датасетов-в-csv-файл-для-дальнейшей-визуализации" data-toc-modified-id="Сохранение-датасетов-в-csv-файл-для-дальнейшей-визуализации-4.6"><span class="toc-item-num">4.6&nbsp;&nbsp;</span>Сохранение датасетов в csv файл для дальнейшей визуализации</a></span></li><li><span><a href="#Ссылка-на-дашборд" data-toc-modified-id="Ссылка-на-дашборд-4.7"><span class="toc-item-num">4.7&nbsp;&nbsp;</span>Ссылка на дашборд</a></span></li><li><span><a href="#Выводы-по-разделу" data-toc-modified-id="Выводы-по-разделу-4.8"><span class="toc-item-num">4.8&nbsp;&nbsp;</span>Выводы по разделу</a></span></li></ul></li><li><span><a href="#Дома-ребенка.-Ресурсы-и-деятельность-организаций-здравоохранения." data-toc-modified-id="Дома-ребенка.-Ресурсы-и-деятельность-организаций-здравоохранения.-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Дома ребенка. Ресурсы и деятельность организаций здравоохранения.</a></span><ul class="toc-item"><li><span><a href="#Предположение" data-toc-modified-id="Предположение-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Предположение</a></span></li><li><span><a href="#Загрузка-и-предобработка-файлов" data-toc-modified-id="Загрузка-и-предобработка-файлов-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Загрузка и предобработка файлов</a></span></li><li><span><a href="#Соединяем-данные-из-всех-папок-в-один-датафрейм" data-toc-modified-id="Соединяем-данные-из-всех-папок-в-один-датафрейм-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>Соединяем данные из всех папок в один датафрейм</a></span></li><li><span><a href="#Ссылка-на-дашборд" data-toc-modified-id="Ссылка-на-дашборд-5.4"><span class="toc-item-num">5.4&nbsp;&nbsp;</span>Ссылка на дашборд</a></span></li><li><span><a href="#Выводы-по-разделу" data-toc-modified-id="Выводы-по-разделу-5.5"><span class="toc-item-num">5.5&nbsp;&nbsp;</span>Выводы по разделу</a></span></li></ul></li><li><span><a href="#Коечный-фонд-по-специализации-в-Субъектах-Российской-Федерации.-Ресурсы-и-деятельность-организаций-здравоохранения." data-toc-modified-id="Коечный-фонд-по-специализации-в-Субъектах-Российской-Федерации.-Ресурсы-и-деятельность-организаций-здравоохранения.-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Коечный фонд по специализации в Субъектах Российской Федерации. Ресурсы и деятельность организаций здравоохранения.</a></span><ul class="toc-item"><li><span><a href="#Предположение" data-toc-modified-id="Предположение-6.1"><span class="toc-item-num">6.1&nbsp;&nbsp;</span>Предположение</a></span></li><li><span><a href="#Загрузка-данных" data-toc-modified-id="Загрузка-данных-6.2"><span class="toc-item-num">6.2&nbsp;&nbsp;</span>Загрузка данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-6.3"><span class="toc-item-num">6.3&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Данные-из-папки-за-2005-год" data-toc-modified-id="Данные-из-папки-за-2005-год-6.3.1"><span class="toc-item-num">6.3.1&nbsp;&nbsp;</span>Данные из папки за 2005 год</a></span></li><li><span><a href="#Данные-из-папки-2007-года" data-toc-modified-id="Данные-из-папки-2007-года-6.3.2"><span class="toc-item-num">6.3.2&nbsp;&nbsp;</span>Данные из папки 2007 года</a></span></li><li><span><a href="#Данные-из-папки-2009-года" data-toc-modified-id="Данные-из-папки-2009-года-6.3.3"><span class="toc-item-num">6.3.3&nbsp;&nbsp;</span>Данные из папки 2009 года</a></span></li><li><span><a href="#Данные-из-папки-за-2011-год" data-toc-modified-id="Данные-из-папки-за-2011-год-6.3.4"><span class="toc-item-num">6.3.4&nbsp;&nbsp;</span>Данные из папки за 2011 год</a></span></li><li><span><a href="#Данные-из-папки-за-2013-год" data-toc-modified-id="Данные-из-папки-за-2013-год-6.3.5"><span class="toc-item-num">6.3.5&nbsp;&nbsp;</span>Данные из папки за 2013 год</a></span></li></ul></li><li><span><a href="#Сохранение-данных" data-toc-modified-id="Сохранение-данных-6.4"><span class="toc-item-num">6.4&nbsp;&nbsp;</span>Сохранение данных</a></span></li><li><span><a href="#Дополнительное-соединение-и-сохранение-датасета" data-toc-modified-id="Дополнительное-соединение-и-сохранение-датасета-6.5"><span class="toc-item-num">6.5&nbsp;&nbsp;</span>Дополнительное соединение и сохранение датасета</a></span></li><li><span><a href="#Ссылка-на-дашборд" data-toc-modified-id="Ссылка-на-дашборд-6.6"><span class="toc-item-num">6.6&nbsp;&nbsp;</span>Ссылка на дашборд</a></span></li><li><span><a href="#Выводы-по-разделу" data-toc-modified-id="Выводы-по-разделу-6.7"><span class="toc-item-num">6.7&nbsp;&nbsp;</span>Выводы по разделу</a></span></li></ul></li><li><span><a href="#Техническое-состояние-и-благоустройство-зданий-лечебно-профилактических-учреждений-здравоохранения.-Ресурсы-и-деятельность-организаций-здравоохранения." data-toc-modified-id="Техническое-состояние-и-благоустройство-зданий-лечебно-профилактических-учреждений-здравоохранения.-Ресурсы-и-деятельность-организаций-здравоохранения.-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>Техническое состояние и благоустройство зданий лечебно-профилактических учреждений здравоохранения. Ресурсы и деятельность организаций здравоохранения.</a></span><ul class="toc-item"><li><span><a href="#Предположение" data-toc-modified-id="Предположение-7.1"><span class="toc-item-num">7.1&nbsp;&nbsp;</span>Предположение</a></span></li><li><span><a href="#Загрузка-данных" data-toc-modified-id="Загрузка-данных-7.2"><span class="toc-item-num">7.2&nbsp;&nbsp;</span>Загрузка данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-7.3"><span class="toc-item-num">7.3&nbsp;&nbsp;</span>Предобработка данных</a></span></li><li><span><a href="#Соединение-в-единый-датасет" data-toc-modified-id="Соединение-в-единый-датасет-7.4"><span class="toc-item-num">7.4&nbsp;&nbsp;</span>Соединение в единый датасет</a></span></li><li><span><a href="#Сохранение-получившихся-датасетов-и-разделение-их-на-стационар-и-амбулатории" data-toc-modified-id="Сохранение-получившихся-датасетов-и-разделение-их-на-стационар-и-амбулатории-7.5"><span class="toc-item-num">7.5&nbsp;&nbsp;</span>Сохранение получившихся датасетов и разделение их на стационар и амбулатории</a></span></li><li><span><a href="#Ссылка-на-дашборд" data-toc-modified-id="Ссылка-на-дашборд-7.6"><span class="toc-item-num">7.6&nbsp;&nbsp;</span>Ссылка на дашборд</a></span></li><li><span><a href="#Вывод-по-разделу" data-toc-modified-id="Вывод-по-разделу-7.7"><span class="toc-item-num">7.7&nbsp;&nbsp;</span>Вывод по разделу</a></span></li></ul></li></ul></div>

# Визуализация медицинской статистики по данным [Росстата](https://rosstat.gov.ru/folder/210/document/13218)

## Описание проекта и цель

Целью проекта является изучение данных сайта Росстата, которые были получены путём парсинга, с дальнейшей визуализацией полученных результатов в виде дашбордов.

## Описание данных

Раздел сайта [Здравоохранение в России](https://rosstat.gov.ru/folder/210/document/13218) содержит сборник данных.  
При подготовке сборника использованы данные, получаемые органами государственной статистики от юридических лиц, населения путем проведения федерального статистического наблюдения, выборочных обследований и других форм статистического наблюдения, данные министерств и ведомств Российской Федерации.  
В сборнике публикуются сведения о медико-демографических аспектах здоровья населения и его отдельных социально-демографических групп, организации лечебно-профилактической помощи и санаторно-курортного лечения населения, а также о состоянии рынка труда в здравоохранении, производстве товаров медицинского назначения, потребительском рынке товаров и услуг здравоохранения. Приведена информация о состоянии окружающей среды и санитарно-эпидемиологическом контроле.  
Статистические данные в разрезе субъектов Российской Федерации начиная с 2015 года выделены в отдельное электронное приложение. Сборник издается начиная с 2001 года с периодичностью один раз в два года, с 2015 года – только в электронном виде.  
В данном проекте делаем дашборды по подразделу **Ресурсы и деятельность организаций здравоохранения**, который имеется в сборнике данных

## Загрузка библиотек

In [1]:
import pandas as pd
import numpy as np
import re

## Число государственных, частных и участковых (сельских) больничных учреждений. Ресурсы и деятельность организаций здравоохранения.

### Предположение

За период 1990 - 2022 число государственных больничных организаций снизилось на 20%, число частных больничных организаций выросло в 4 раза, а количество участковых (сельских) больниц уменьшилось в 20 раз.

### Загрузка и предобработка данных

In [2]:
hospital_facilities_2005 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2005/Ресурсы и деятельность учреждений здравоохранения/больничные учреждения.csv')

hospital_facilities_2005.columns = hospital_facilities_2005.iloc[0]

hospital_facilities_2005.rename(columns = {np.nan:'name'}, inplace=True)

hospital_facilities_2005 = hospital_facilities_2005.drop(index = [0, 1]).reset_index(drop=True)

hospital_facilities_2005.head()

0                                    name    1990    1995    2000    2001  \
0             Число больничных учреждений   12762   12064   10704   10561   
1                из них негосударственные     ...      29      87     105   
2    Число коек в больничных учреждениях:     NaN     NaN     NaN     NaN   
3                             всего, тыс.  2037.6  1850.5  1671.6  1653.4   
4  из них в негосударственных учреждениях     ...     2.5     7.7     7.9   

0    2002    2003    2004  
0   10320   10101    9851  
1      97     111     122  
2     NaN     NaN     NaN  
3  1619.7  1596.6  1600.7  
4     7.8     7.9     9.8

Посмотрим на уникальные названия колонки name, так как после изучения каждой таблицы "больничные учреждения" из каждой папки по годам, будем всё это соединять в единую таблицу по колонкам. Колонка name может содержать неодинаковые (неунифицированные) названия

In [3]:
hospital_facilities_2005['name'].unique()

array(['Число больничных учреждений', 'из них негосударственные',
       'Число коек в больничных учреждениях:', 'всего, тыс.',
       'из них в негосударственных учреждениях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan,
       'Число центральных районних больниц'], dtype=object)

In [4]:
hospital_facilities_2007 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2007/Ресурсы и деятельность учреждений здравоохранения/больничные учреждения.csv')

hospital_facilities_2007.columns = hospital_facilities_2007.iloc[0]

hospital_facilities_2007.rename(columns = {np.nan:'name'}, inplace=True)

hospital_facilities_2007 = hospital_facilities_2007.drop(index = [0, 1]).reset_index(drop=True)

hospital_facilities_2007.head()

0                                  name    1990    1995    2000    2005  \
0           Число больничных учреждений   12762   12064   10704    9479   
1              из них негосударственные     ...      29      87     293   
2    из них частной формы собственности     ...     ...     ...     ...   
3  Число коек в больничных учреждениях:     NaN     NaN     NaN     NaN   
4                           всего, тыс.  2037.6  1850.5  1671.6  1575.4   

0    2006  
0    7478  
1     320  
2     148  
3     NaN  
4  1553.6

In [5]:
hospital_facilities_2007['name'].unique()

array(['Число больничных учреждений', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных учреждениях:', 'всего, тыс.',
       'из них в негосударственных  учреждениях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной  больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan,
       'из них частной формы  собственности'], dtype=object)

In [6]:
hospital_facilities_2009 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2009/Ресурсы и деятельность учреждений здравоохранения/больничные учреждения.csv')

hospital_facilities_2009.columns = hospital_facilities_2009.iloc[0]

hospital_facilities_2009.rename(columns = {np.nan:'name'}, inplace=True)

hospital_facilities_2009 = hospital_facilities_2009.drop(index = [0, 1]).reset_index(drop=True)

hospital_facilities_2009.head()

0                                  name    1995    2000    2005    2007  \
0           Число больничных учреждений   12064   10704    9479    6777   
1              из них негосударственные      29      87     293     333   
2   из них частной формы  собственности     ...     ...     ...     177   
3  Число коек в больничных учреждениях:     NaN     NaN     NaN     NaN   
4                           всего, тыс.  1850.5  1671.6  1575.4  1522.1   

0     2008  
0     6545  
1      253  
2      120  
3      NaN  
4  1398,52

In [7]:
hospital_facilities_2009['name'].unique()

array(['Число больничных учреждений', 'из них негосударственные',
       'из них частной формы  собственности',
       'Число коек в больничных учреждениях:', 'всего, тыс.',
       'из них в негосударственных  учреждениях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной  больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

In [8]:
hospital_facilities_2011 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2011/Ресурсы и деятельность учреждений здравоохранения/больничные учреждения.csv')

hospital_facilities_2011.columns = hospital_facilities_2011.iloc[0]

hospital_facilities_2011.rename(columns = {np.nan:'name'}, inplace=True)

hospital_facilities_2011 = hospital_facilities_2011.drop(index = [0, 1]).reset_index(drop=True)

hospital_facilities_2011.head()

0                                  name    1995    2000    2005    2009  \
0           Число больничных учреждений   12064   10704    9479    6454   
1              из них негосударственные      29      87     293     224   
2    из них частной формы собственности       -       -     ...     107   
3  Число коек в больничных учреждениях:     NaN     NaN     NaN     NaN   
4                           всего, тыс.  1850.5  1671.6  1575.4  1373.4   

0    2010  
0    6308  
1     224  
2     115  
3     NaN  
4  1339.5

In [9]:
hospital_facilities_2011['name'].unique()

array(['Число больничных учреждений', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных учреждениях:', 'всего, тыс.',
       'из них в негосударственных учреждениях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

In [10]:
hospital_facilities_2013 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2013/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.1. больничные организации.csv')

hospital_facilities_2013 = hospital_facilities_2013.drop(index = [0]).reset_index(drop=True)

hospital_facilities_2013.rename(columns = {'Unnamed: 0':'name'}, inplace=True)

hospital_facilities_2013.head()

name    2000    2005     2010     2011  \
0           Число больничных организаций   10704    9479     6308     6343   
1               из них негосударственные      87     293      224      220   
2     из них частной формы собственности       -       …      115      116   
3  Число коек в больничных организациях:     NaN     NaN      NaN      NaN   
4                            всего, тыс.  1671,6  1575,4  1339,51  1347,11   

      2012  
0     6172  
1      225  
2      127  
3      NaN  
4  1332,31

In [11]:
hospital_facilities_2013['name'].unique()

array(['Число больничных организаций', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных организациях:', 'всего, тыс.',
       'из них в негосударственных организациях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

In [12]:
hospital_facilities_2011['name'].unique() == hospital_facilities_2013['name'].unique()

array([False,  True,  True, False,  True, False,  True,  True,  True,
        True,  True,  True, False])

In [13]:
print(hospital_facilities_2011['name'][[0, 3, 5]])

print('-'*50)

print(hospital_facilities_2013['name'][[0, 3, 5]])

0               Число больничных учреждений
3      Число коек в больничных учреждениях:
5    из них в негосударственных учреждениях
Name: name, dtype: object
--------------------------------------------------
0               Число больничных организаций
3      Число коек в больничных организациях:
5    из них в негосударственных организациях
Name: name, dtype: object


При сравнению с названиями колонки name 2011 года с 2013 есть несовпадающие значения. Видно, что слово "учреждения" изменили на "организации".

Создадим новую переменную `res` в которой будет словарь со старыми значениями из строк колонки "name" 2013 года и новые значения из строк колонки "name" 2011 года.

In [14]:
res = {list(hospital_facilities_2013['name'])[i]: list(hospital_facilities_2011['name'])[i] for i in range(len(list(hospital_facilities_2013['name'])))}

res

{'Число больничных организаций': 'Число больничных учреждений',
 'из них негосударственные': 'из них негосударственные',
 'из них частной формы собственности': 'из них частной формы собственности',
 'Число коек в больничных организациях:': 'Число коек в больничных учреждениях:',
 'всего, тыс.': 'всего, тыс.',
 'из них в негосударственных организациях': 'из них в негосударственных учреждениях',
 'на 10 000 человек населения': 'на 10 000 человек населения',
 'Число центральных районных больниц': 'Число центральных районных больниц',
 'в них коек, тыс.': 'в них коек, тыс.',
 'средняя мощность одной больницы, коек': 'средняя мощность одной больницы, коек',
 'Число районных больниц': 'Число районных больниц',
 'Число участковых больниц': 'Число участковых больниц',
 nan: nan}

In [15]:
hospital_facilities_2013['name'] = hospital_facilities_2013['name'].map(res)

hospital_facilities_2013['name']

0                Число больничных учреждений
1                   из них негосударственные
2         из них частной формы собственности
3       Число коек в больничных учреждениях:
4                                всего, тыс.
5     из них в негосударственных учреждениях
6         из них частной формы собственности
7                на 10 000 человек населения
8         Число центральных районных больниц
9                           в них коек, тыс.
10     средняя мощность одной больницы, коек
11                    Число районных больниц
12                          в них коек, тыс.
13     средняя мощность одной больницы, коек
14                  Число участковых больниц
15                          в них коек, тыс.
16     средняя мощность одной больницы, коек
17                                       NaN
18               Число больничных учреждений
19                  из них негосударственные
20        из них частной формы собственности
21      Число коек в больничных учреждениях:
22        

In [16]:
hospital_facilities_2015 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2015/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.1. больничные организации.csv')

hospital_facilities_2015 = hospital_facilities_2015.drop(index = [0]).reset_index(drop=True)

hospital_facilities_2015.rename(columns = {'Unnamed: 0':'name'}, inplace=True)

hospital_facilities_2015.head()

name    2005     2010     2011     2012  \
0           Число больничных организаций    9479     6308     6343     6172   
1               из них негосударственные     293      224      220      225   
2     из них частной формы собственности       …      115      116      127   
3  Число коек в больничных организациях:     NaN      NaN      NaN      NaN   
4                            всего, тыс.  1575,4  1339,51  1347,11  1332,31   

      2013     2014  
0     5870     5638  
1      233      246  
2      143      166  
3      NaN      NaN  
4  1301,91  1266,81

In [17]:
hospital_facilities_2015['name'].unique()

array(['Число больничных организаций', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных организациях:', 'всего, тыс.',
       'из них в негосударственных организациях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

В 2015 году также видно, что слово "учреждения" заменили на "организации", восплользуемся ранее созданным словарём из переменной `res` и изменим названия для унификации.

In [18]:
hospital_facilities_2015['name'] = hospital_facilities_2015['name'].map(res)

hospital_facilities_2015['name']

0                Число больничных учреждений
1                   из них негосударственные
2         из них частной формы собственности
3       Число коек в больничных учреждениях:
4                                всего, тыс.
5     из них в негосударственных учреждениях
6         из них частной формы собственности
7                на 10 000 человек населения
8         Число центральных районных больниц
9                           в них коек, тыс.
10     средняя мощность одной больницы, коек
11                    Число районных больниц
12                          в них коек, тыс.
13     средняя мощность одной больницы, коек
14                  Число участковых больниц
15                          в них коек, тыс.
16     средняя мощность одной больницы, коек
17                                       NaN
18               Число больничных учреждений
19                  из них негосударственные
20        из них частной формы собственности
21      Число коек в больничных учреждениях:
22        

In [19]:
hospital_facilities_2017 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2017/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.1. больничные организации.csv')

hospital_facilities_2017 = hospital_facilities_2017.drop(index = [0]).reset_index(drop=True)

hospital_facilities_2017.rename(columns = {'Unnamed: 0':'name'}, inplace=True)

hospital_facilities_2017.head()

name    2005     2010     2013     2014  \
0           Число больничных организаций    9479     6308     5870     5638   
1               из них негосударственные     293      224      233      246   
2     из них частной формы собственности       …      115      143      166   
3  Число коек в больничных организациях:     NaN      NaN      NaN      NaN   
4                            всего, тыс.  1575,4  1339,51  1301,91  1266,81   

     2015    2016  
0    5433    5357  
1     245     266  
2     180     205  
3     NaN     NaN  
4  1222,0  1197,2

In [20]:
hospital_facilities_2017['name'].unique()

array(['Число больничных организаций', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных организациях:', 'всего, тыс.',
       'из них в негосударственных организациях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

Изменим названия строк колонки name 2017 с применением ранее созданного словаря в переменной `res`

In [21]:
hospital_facilities_2017['name'] = hospital_facilities_2017['name'].map(res)

In [22]:
hospital_facilities_2017

name                    2005  \
0              Число больничных учреждений                    9479   
1                 из них негосударственные                     293   
2       из них частной формы собственности                       …   
3     Число коек в больничных учреждениях:                     NaN   
4                              всего, тыс.                  1575,4   
5   из них в негосударственных учреждениях                    32,4   
6       из них частной формы собственности                       …   
7              на 10 000 человек населения                   110,9   
8       Число центральных районных больниц                    1734   
9                         в них коек, тыс.                   345,1   
10   средняя мощность одной больницы, коек                     199   
11                  Число районных больниц                     267   
12                        в них коек, тыс.                    30,8   
13   средняя мощность одной больницы, коек                     115   
14                Число участковых больниц                    2631   
15                        в них коек, тыс.                    62,3   
16   средняя мощность одной больницы, коек                      24   
17                                     NaN  В городских поселениях   
18             Число больничных учреждений                    5820   
19                из них негосударственные                     286   
20      из них частной формы собственности                       …   
21    Число коек в больничных учреждениях:                     NaN   
22                             всего, тыс.                  1365,9   
23  из них в негосударственных учреждениях                    32,1   
24      из них частной формы собственности                       …   
25             на 10 000 человек населения                   130,3   
26      Число центральных районных больниц                    1046   
27                  Число районных больниц                     148   
28                Число участковых больниц                      40   
29                                     NaN    В сельской местности   
30             Число больничных учреждений                    3659   
31                из них негосударственные                       7   
32      из них частной формы собственности                       …   
33    Число коек в больничных учреждениях:                     NaN   
34                             всего, тыс.                   209,5   
35  из них в негосударственных учреждениях                     0,3   
36      из них частной формы собственности                       …   
37             на 10 000 человек населения                    55,6   
38      Число центральных районных больниц                     688   
39                  Число районных больниц                     119   
40                Число участковых больниц                    2591   

                      2010                    2013                    2014  \
0                     6308                    5870                    5638   
1                      224                     233                     246   
2                      115                     143                     166   
3                      NaN                     NaN                     NaN   
4                  1339,51                 1301,91                 1266,81   
5                     21,7                    23,2                    24,9   
6                      4,1                     7,9                    11,0   
7                     93,8                    90,6                    86,6   
8                     1754                    1690                    1470   
9                    327,1                   267,0                   224,8   
10                     186                   158                       153   
11                     191                     206                     417   
12                    24,2                    23,7                    54,1   
13   

In [23]:
hospital_facilities_2019 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2019/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.1. больничные организации.csv')

hospital_facilities_2019 = hospital_facilities_2019.drop(index = [0]).reset_index(drop=True)

hospital_facilities_2019.rename(columns = {'Unnamed: 0':'name'}, inplace=True)

hospital_facilities_2019.head()

name    2005     2010     2015     2016  \
0           Число больничных организаций    9479     6308     5433     5357   
1               из них негосударственные     293      224      245      266   
2     из них частной формы собственности       …      115      180      205   
3  Число коек в больничных организациях:     NaN      NaN      NaN      NaN   
4                            всего, тыс.  1575,4  1339,51  1222,01  1197,21   

      2017     2018  
0     5293     5257  
1      294      319  
2      241      259  
3      NaN      NaN  
4  1182,71  1172,81

In [24]:
hospital_facilities_2019['name'].unique()

array(['Число больничных организаций', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных организациях:', 'всего, тыс.',
       'из них в негосударственных организациях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

Изменим названия строк колонки name папки 2019 года с применением ранее созданного словаря в переменной `res`

In [25]:
hospital_facilities_2019['name'] = hospital_facilities_2019['name'].map(res)

In [26]:
hospital_facilities_2019

name                    2005  \
0              Число больничных учреждений                    9479   
1                 из них негосударственные                     293   
2       из них частной формы собственности                       …   
3     Число коек в больничных учреждениях:                     NaN   
4                              всего, тыс.                  1575,4   
5   из них в негосударственных учреждениях                    32,4   
6       из них частной формы собственности                       …   
7              на 10 000 человек населения                   110,9   
8       Число центральных районных больниц                    1734   
9                         в них коек, тыс.                   345,1   
10   средняя мощность одной больницы, коек                     199   
11                  Число районных больниц                     267   
12                        в них коек, тыс.                    30,8   
13   средняя мощность одной больницы, коек                     115   
14                Число участковых больниц                    2631   
15                        в них коек, тыс.                    62,3   
16   средняя мощность одной больницы, коек                      24   
17                                     NaN  В городских поселениях   
18             Число больничных учреждений                    5820   
19                из них негосударственные                     286   
20      из них частной формы собственности                       …   
21    Число коек в больничных учреждениях:                     NaN   
22                             всего, тыс.                  1365,9   
23  из них в негосударственных учреждениях                    32,1   
24      из них частной формы собственности                       …   
25             на 10 000 человек населения                   130,3   
26      Число центральных районных больниц                    1046   
27                  Число районных больниц                     148   
28                Число участковых больниц                      40   
29                                     NaN    В сельской местности   
30             Число больничных учреждений                    3659   
31                из них негосударственные                       7   
32      из них частной формы собственности                       …   
33    Число коек в больничных учреждениях:                     NaN   
34                             всего, тыс.                   209,5   
35  из них в негосударственных учреждениях                     0,3   
36      из них частной формы собственности                       …   
37             на 10 000 человек населения                    55,6   
38      Число центральных районных больниц                     688   
39                  Число районных больниц                     119   
40                Число участковых больниц                    2591   

                      2010                    2015                    2016  \
0                     6308                    5433                    5357   
1                      224                     245                     266   
2                      115                     180                     205   
3                      NaN                     NaN                     NaN   
4                  1339,51                 1222,01                 1197,21   
5                     21,7                    24,1                    23,5   
6                      4,1                    12,8                    12,9   
7                     93,8                    83,4                    81,6   
8                     1754                    1426                    1388   
9                    327,1                   205,6                   192,6   
10                     186                     144                     139   
11                     191                     415                     425   
12                    24,2                    54,5                    58,1   
13   

In [27]:
hospital_facilities_2021 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2021/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.1. больничные организации.csv')

hospital_facilities_2021 = hospital_facilities_2021.drop(index = [0]).reset_index(drop=True)

hospital_facilities_2021.rename(columns = {'Unnamed: 0':'name'}, inplace=True)

hospital_facilities_2021.head()

name     2005      2010      2015  \
0           Число больничных организаций    9 479     6 308     5 433   
1               из них негосударственные      293       224       245   
2     из них частной формы собственности        …       115       180   
3  Число коек в больничных организациях:      NaN       NaN       NaN   
4                            всего, тыс.  1 575,4  1 339,51  1 222,01   

       2018      2019      2020  
0     5 257     5 130     5 065  
1       319       328       344  
2       259       274       294  
3       NaN       NaN       NaN  
4  1 172,81  1 173,61  1 188,81

In [28]:
hospital_facilities_2021['name'].unique()

array(['Число больничных организаций', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных организациях:', 'всего, тыс.',
       'из них в негосударственных организациях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

In [29]:
hospital_facilities_2021['name'] = hospital_facilities_2021['name'].map(res)

In [30]:
hospital_facilities_2021

name                    2005  \
0              Число больничных учреждений                   9 479   
1                 из них негосударственные                     293   
2       из них частной формы собственности                       …   
3     Число коек в больничных учреждениях:                     NaN   
4                              всего, тыс.                 1 575,4   
5   из них в негосударственных учреждениях                    32,4   
6       из них частной формы собственности                       …   
7              на 10 000 человек населения                   110,9   
8       Число центральных районных больниц                   1 734   
9                         в них коек, тыс.                   345,1   
10   средняя мощность одной больницы, коек                     199   
11                  Число районных больниц                     267   
12                        в них коек, тыс.                    30,8   
13   средняя мощность одной больницы, коек                     115   
14                Число участковых больниц                   2 631   
15                        в них коек, тыс.                    62,3   
16   средняя мощность одной больницы, коек                      24   
17                                     NaN  В городских поселениях   
18             Число больничных учреждений                   5 820   
19                из них негосударственные                     286   
20      из них частной формы собственности                       …   
21    Число коек в больничных учреждениях:                     NaN   
22                             всего, тыс.                 1 365,9   
23  из них в негосударственных учреждениях                    32,1   
24      из них частной формы собственности                       …   
25             на 10 000 человек населения                   130,3   
26      Число центральных районных больниц                   1 046   
27                  Число районных больниц                     148   
28                Число участковых больниц                      40   
29                                     NaN    В сельской местности   
30             Число больничных учреждений                   3 659   
31                из них негосударственные                       7   
32      из них частной формы собственности                       …   
33    Число коек в больничных учреждениях:                     NaN   
34                             всего, тыс.                   209,5   
35  из них в негосударственных учреждениях                     0,3   
36      из них частной формы собственности                       …   
37             на 10 000 человек населения                    55,6   
38      Число центральных районных больниц                     688   
39                  Число районных больниц                     119   
40                Число участковых больниц                   2 591   

                      2010                    2015                    2018  \
0                    6 308                   5 433                   5 257   
1                      224                     245                     319   
2                      115                     180                     259   
3                      NaN                     NaN                     NaN   
4                 1 339,51                1 222,01                1 172,81   
5                     21,7                    24,1                    27,0   
6                      4,1                    12,8                    17,3   
7                     93,8                    83,4                    79,9   
8                    1 754                   1 426                   1 361   
9                    327,1                   205,6                   183,7   
10                     186                     144                     135   
11                     191                     415                     418   
12                    24,2                    54,5                    55,8   
13   

In [31]:
hospital_facilities_2023 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2023/БОЛЬНИЧНЫЕ ОРГАНИЗАЦИИ - 3.1.csv')

hospital_facilities_2023.columns = hospital_facilities_2023.iloc[0]

hospital_facilities_2023 = hospital_facilities_2023.drop(index = [0, 1]).reset_index(drop=True)

hospital_facilities_2023.rename(columns = {np.nan:'name'}, inplace=True)

hospital_facilities_2023.head()

0                                   name      2010      2015      2019  \
0           Число больничных организаций     6 308     5 433     5 130   
1               из них негосударственные       224       245       328   
2     из них частной формы собственности       115       180       274   
3  Число коек в больничных организациях:       NaN       NaN       NaN   
4                            всего, тыс.  1 339,51  1 222,01  1 173,61   

0      2020      2021       2022  
0     5 065     5 072      5 167  
1       344       410        458  
2       294       367        409  
3       NaN       NaN        NaN  
4  1 188,81  1 162,11   1 141,91

In [32]:
hospital_facilities_2023['name'].unique()

array(['Число больничных организаций', 'из них негосударственные',
       'из них частной формы собственности',
       'Число коек в больничных организациях:', 'всего, тыс.',
       'из них в негосударственных организациях',
       'на 10 000 человек населения',
       'Число центральных районных больниц', 'в них коек, тыс.',
       'средняя мощность одной больницы, коек', 'Число районных больниц',
       'Число участковых больниц', nan], dtype=object)

In [33]:
hospital_facilities_2023['name'] = hospital_facilities_2023['name'].map(res)

hospital_facilities_2023

0                                     name                    2010      2015  \
0              Число больничных учреждений                   6 308     5 433   
1                 из них негосударственные                     224       245   
2       из них частной формы собственности                     115       180   
3     Число коек в больничных учреждениях:                     NaN       NaN   
4                              всего, тыс.                1 339,51  1 222,01   
5   из них в негосударственных учреждениях                    21,7      24,1   
6       из них частной формы собственности                     4,1      12,8   
7              на 10 000 человек населения                    93,8      83,4   
8       Число центральных районных больниц                   1 754     1 426   
9                         в них коек, тыс.                   327,1     205,6   
10   средняя мощность одной больницы, коек                     186       144   
11                  Число районных больниц                     191       415   
12                        в них коек, тыс.                    24,2      54,5   
13   средняя мощность одной больницы, коек                     126       131   
14                Число участковых больниц                     400        69   
15                        в них коек, тыс.                    11,2       2,2   
16   средняя мощность одной больницы, коек                      28        32   
17                                     NaN  В городских поселениях       NaN   
18             Число больничных учреждений                   4 959     4 397   
19                из них негосударственные                     216       239   
20      из них частной формы собственности                     110       176   
21    Число коек в больничных учреждениях:                     NaN       NaN   
22                             всего, тыс.                1 186,11  1 060,11   
23  из них в негосударственных учреждениях                    20,2      21,8   
24      из них частной формы собственности                     3,4      12,1   
25             на 10 000 человек населения                   112,5      97,6   
26      Число центральных районных больниц                   1 027       829   
27                  Число районных больниц                     112       230   
28                Число участковых больниц                      18         4   
29                                     NaN    В сельской местности       NaN   
30             Число больничных учреждений                   1 349     1 036   
31                из них негосударственные                       8         6   
32      из них частной формы собственности                       5         4   
33    Число коек в больничных учреждениях:                     NaN       NaN   
34                             всего, тыс.                  153,41    161,91   
35  из них в негосударственных учреждениях                     1,6       2,3   
36      из них частной формы собственности                    0,68      0,60   
37             на 10 000 человек населения                    40,9      42,7   
38      Число центральных районных больниц                     727       597   
39                  Число районных больниц                      79       185   
40                Число участковых больниц                     382        65   

0       2019      2020      2021       2022  
0      5 130     5 065     5 072      5 167  
1        328       344       410        458  
2        274       294       367        409  
3        NaN       NaN       NaN        NaN  
4   1 173,61  1 188,81  1 162,11   1 141,91  
5       29,8      32,7      32,6       36,1  
6       20,5      23,3      24,9       28,0  
7       80,0      81,3       NaN        NaN  
8      1 325     1 316     1 311      1 280  
9      177,4     175,4     171,6        164  
10       134       133       131        128  
11       416       409       391        396  
12      52,6      51,5      49,1       48

In [34]:
hospital_facilities_2005

0                                     name                    1990  \
0              Число больничных учреждений                   12762   
1                 из них негосударственные                     ...   
2     Число коек в больничных учреждениях:                     NaN   
3                              всего, тыс.                  2037.6   
4   из них в негосударственных учреждениях                     ...   
5              на 10 000 человек населения                   137.4   
6       Число центральных районных больниц                    1798   
7                         в них коек, тыс.                   451.0   
8    средняя мощность одной больницы, коек                     251   
9                   Число районных больниц                     324   
10                        в них коек, тыс.                    33.8   
11   средняя мощность одной больницы, коек                     104   
12                Число участковых больниц                    4813   
13                        в них коек, тыс.                   156.3   
14   средняя мощность одной больницы, коек                      32   
15                                     NaN  В городских поселениях   
16             Число больничных учреждений                     ...   
17                из них негосударственные                     ...   
18    Число коек в больничных учреждениях:                     NaN   
19                             всего, тыс.                     ...   
20  из них в негосударственных учреждениях                     ...   
21             на 10 000 человек населения                     ...   
22      Число центральных районних больниц                    1196   
23                  Число районных больниц                     146   
24                Число участковых больниц                       -   
25                                     NaN    В сельской местности   
26             Число больничных учреждений                     ...   
27                из них негосударственные                     ...   
28    Число коек в больничных учреждениях:                     NaN   
29                             всего, тыс.                     ...   
30  из них в негосударственных учреждениях                     ...   
31             на 10 000 человек населения                     ...   
32      Число центральных районных больниц                     602   
33                  Число районных больниц                     178   
34                Число участковых больниц                    4813   

0                     1995                    2000                    2001  \
0                    12064                   10704                   10561   
1                       29                      87                     105   
2                      NaN                     NaN                     NaN   
3                   1850.5                  1671.6                  1653.4   
4                      2.5                     7.7                     7.9   
5                    125.8                   115.0                   114.4   
6                     1786                    1765                    1756   
7                    409.7                   373.6                   365.5   
8                      229                     211                     208   
9                      294                     263                     266   
10                    31.6                    28.4                    29.2   
11                     107                     107                     109   
12                    4409                    3310                    3290   
13                   129.0                    85.1                    83.4   
14                      29                      26                      25   
15  В городских поселениях  В городских поселениях  В городских поселениях   
16                    6627                    6326                    6227   
17                      29                      87                     104 

In [35]:
hospital_facilities_2007

0                                      name                    1990  \
0               Число больничных учреждений                   12762   
1                  из них негосударственные                     ...   
2        из них частной формы собственности                     ...   
3      Число коек в больничных учреждениях:                     NaN   
4                               всего, тыс.                  2037.6   
5   из них в негосударственных  учреждениях                     ...   
6        из них частной формы собственности                     ...   
7               на 10 000 человек населения                   137.4   
8        Число центральных районных больниц                    1798   
9                          в них коек, тыс.                   451.0   
10   средняя мощность одной  больницы, коек                     251   
11                   Число районных больниц                     324   
12                         в них коек, тыс.                    33.8   
13   средняя мощность одной  больницы, коек                     104   
14                 Число участковых больниц                    4813   
15                         в них коек, тыс.                   156.3   
16   средняя мощность одной  больницы, коек                      32   
17                                      NaN  В городских поселениях   
18              Число больничных учреждений                     ...   
19                 из них негосударственные                     ...   
20      из них частной формы  собственности                     ...   
21     Число коек в больничных учреждениях:                     NaN   
22                              всего, тыс.                     ...   
23  из них в негосударственных  учреждениях                     ...   
24      из них частной формы  собственности                     ...   
25              на 10 000 человек населения                     ...   
26       Число центральных районных больниц                    1196   
27                   Число районных больниц                     146   
28                 Число участковых больниц                       -   
29                                      NaN    В сельской местности   
30              Число больничных учреждений                     ...   
31                 из них негосударственные                     ...   
32      из них частной формы  собственности                     ...   
33     Число коек в больничных учреждениях:                     NaN   
34                              всего, тыс.                     ...   
35  из них в негосударственных  учреждениях                     ...   
36       из них частной формы собственности                     ...   
37              на 10 000 человек населения                     ...   
38       Число центральных районных больниц                     602   
39                   Число районных больниц                     178   
40                 Число участковых больниц                    4813   

0                     1995                    2000                    2005  \
0                    12064                   10704                    9479   
1                       29                      87                     293   
2                      ...                     ...                     ...   
3                      NaN                     NaN                     NaN   
4                   1850.5                  1671.6                  1575.4   
5                      2.5                     7.7                    32.4   
6                      ...                     ...                     ...   
7                    125.8                   115.0                   111.3   
8                     1786                    1765                    1734   
9                    409.7                   373.6                   345.1   
10                     229                     211                     199   
11                     294                     263                     267   
12 

In [ ]:
hospital_facilities_2009

0                                      name                    1995  \
0               Число больничных учреждений                   12064   
1                  из них негосударственные                      29   
2       из них частной формы  собственности                     ...   
3      Число коек в больничных учреждениях:                     NaN   
4                               всего, тыс.                  1850.5   
5   из них в негосударственных  учреждениях                     2.5   
6       из них частной формы  собственности                     ...   
7               на 10 000 человек населения                   125.8   
8        Число центральных районных больниц                    1786   
9                          в них коек, тыс.                   409.7   
10   средняя мощность одной  больницы, коек                     229   
11                   Число районных больниц                     294   
12                         в них коек, тыс.                    31.6   
13   средняя мощность одной  больницы, коек                     107   
14                 Число участковых больниц                    4409   
15                         в них коек, тыс.                   129.0   
16   средняя мощность одной  больницы, коек                      29   
17                                      NaN  В городских поселениях   
18              Число больничных учреждений                    6627   
19                 из них негосударственные                      29   
20      из них частной формы  собственности                     ...   
21     Число коек в больничных учреждениях:                     NaN   
22                              всего, тыс.                  1569.2   
23  из них в негосударственных  учреждениях                       -   
24      из них частной формы  собственности                     ...   
25              на 10 000 человек населения                   145.5   
26       Число центральных районных больниц                    1103   
27                   Число районных больниц                     143   
28                 Число участковых больниц                      11   
29                                      NaN    В сельской местности   
30              Число больничных учреждений                    5437   
31                 из них негосударственные                       -   
32      из них частной формы  собственности                     ...   
33     Число коек в больничных учреждениях:                     NaN   
34                              всего, тыс.                   281.3   
35  из них в негосударственных  учреждениях                       -   
36      из них частной формы  собственности                     ...   
37              на 10 000 человек населения                    71.7   
38       Число центральных районных больниц                     683   
39                   Число районных больниц                     151   
40                 Число участковых больниц                    4398   

0                     2000                    2005                    2007  \
0                    10704                    9479                    6777   
1                       87                     293                     333   
2                      ...                     ...                     177   
3                      NaN                     NaN                     NaN   
4                   1671.6                  1575.4                  1522.1   
5                      7.7                    32.4                    30.3   
6                      ...                     ...                     5.1   
7                    115.0                   111.3                   107.2   
8                     1765                    1734                    1747   
9                    373.6                   345.1                   380.6   
10                     211                     199                     218   
11                     263                     267                     215   
12 

In [ ]:
hospital_facilities_2011

In [ ]:
hospital_facilities_2013

In [ ]:
hospital_facilities_2015

In [ ]:
hospital_facilities_2017

In [ ]:
hospital_facilities_2019

In [ ]:
hospital_facilities_2021

In [ ]:
hospital_facilities_2023

### Соединение в общий датасет

Так как число строк меньше только в данных из папки за 2005 год, остальные все данные по годам имеют одинаковое кол-во заполненных строк, то начнём конкатенацию с данных из папки 2007. Данные из папки 2005 присоединим в самом конце.

In [ ]:
hospital_facilities_1 = pd.concat([hospital_facilities_2007, hospital_facilities_2009], axis=1)

hospital_facilities_1

In [ ]:
hospital_facilities_2 = pd.concat([hospital_facilities_1, hospital_facilities_2011], axis=1)

hospital_facilities_2.head()

In [ ]:
hospital_facilities_3 = pd.concat([hospital_facilities_2, hospital_facilities_2013], axis=1)

hospital_facilities_3.head()

In [ ]:
hospital_facilities_4 = pd.concat([hospital_facilities_3, hospital_facilities_2015], axis=1)

hospital_facilities_4.head()

In [ ]:
hospital_facilities_5 = pd.concat([hospital_facilities_4, hospital_facilities_2017], axis=1)

hospital_facilities_5.head()

In [ ]:
hospital_facilities_6 = pd.concat([hospital_facilities_5, hospital_facilities_2019], axis=1)

hospital_facilities_6.head()

In [ ]:
hospital_facilities_7 = pd.concat([hospital_facilities_6, hospital_facilities_2021], axis=1)

hospital_facilities_7.head()

In [ ]:
hospital_facilities_8 = pd.concat([hospital_facilities_7, hospital_facilities_2023], axis=1)

hospital_facilities_8.head()

Теперь удалим столбцы дубликаты по названию

In [ ]:
hospital_facilities_8 = hospital_facilities_8.loc[:, ~hospital_facilities_8.columns.duplicated()].copy()

In [ ]:
hospital_facilities_8

Теперь присоединим ко всем данным, данные из папки за 2005 год

In [ ]:
hospital_facilities_total = hospital_facilities_8.merge(hospital_facilities_2005, how='left')

hospital_facilities_total

После соединения уберёмя явные полные дубликаты

In [ ]:
hospital_facilities_total = hospital_facilities_total.drop_duplicates().reset_index(drop=True)

Теперь заменим запятые между цифрами на точку для корректного отображения "плавающей" точки в программе по визуализации и построению дашборов.

In [ ]:
hospital_facilities_total = hospital_facilities_total.apply(lambda x: x.str.replace(',', '.'))

In [ ]:
hospital_facilities_total

Для корректного отображения данных с "плавающей" точкой удалим неразрывный пробел во всех строках датафрейма

In [ ]:
hospital_facilities_total.iloc[:,1:] = hospital_facilities_total.iloc[:,1:].apply(lambda x: x.str.replace('\xa0', ''))

### Создание датасетов

Создадим датафреймы где будут: 
- общие значения;
- городские поселения;
- сельская местность.

#### Датасет с общими значениями

In [ ]:
hospital_facilities_total_copy = hospital_facilities_total.iloc[:18,:].copy()

hospital_facilities_total_copy

#### Датасет городских поселений

In [ ]:
hospital_facilities_urban_settlements = hospital_facilities_total.iloc[19:30,:].copy()

hospital_facilities_urban_settlements

#### Датасет сельской местности

In [ ]:
hospital_facilities_rural_area = hospital_facilities_total.iloc[31:,:].copy()

hospital_facilities_rural_area

### Транспонирование ранее созданных 3-х датасетов

Далее транспорнируем все три датасета, чтобы индексами были года, а колонками наименования.

#### Транспонированный датасет с общими значениями 

In [ ]:
hospital_facilities_total_copy = hospital_facilities_total_copy.T

hospital_facilities_total_copy

Теперь сделаем первую строку названиями колонок

In [ ]:
hospital_facilities_total_copy.columns = list(hospital_facilities_total_copy.iloc[0])

hospital_facilities_total_copy = hospital_facilities_total_copy.drop(index = 'name')

hospital_facilities_total_copy

#### Транспонированный датасет с городскими поселениями

In [ ]:
hospital_facilities_urban_settlements = hospital_facilities_urban_settlements.T

# изменение первой строки в названия столбцов

hospital_facilities_urban_settlements.columns = list(hospital_facilities_urban_settlements.iloc[0])

hospital_facilities_urban_settlements = hospital_facilities_urban_settlements.drop(index = 'name')

hospital_facilities_urban_settlements

#### Транспонированный датасет сельской местности

In [ ]:
hospital_facilities_rural_area = hospital_facilities_rural_area.T

# изменение первой строки в название столбцов

hospital_facilities_rural_area.columns = list(hospital_facilities_rural_area.iloc[0])

hospital_facilities_rural_area = hospital_facilities_rural_area.drop(index = 'name')

In [ ]:
hospital_facilities_rural_area

### Сохранение датасетов в csv файл для дальнейшей визуализации

In [ ]:
hospital_facilities_total_copy.to_csv('hospital_facilities_total_copy.csv')

In [ ]:
hospital_facilities_urban_settlements.to_csv('hospital_facilities_urban_settlements.csv')

In [ ]:
hospital_facilities_rural_area.to_csv('hospital_facilities_rural_area.csv')

### Ссылка на дашборд

Дашборд выполнен в программе Tableau и доступен по [ссылке](https://public.tableau.com/views/hospital_facilities_corrected/sheet16?:language=en-US&:sid=&:display_count=n&:origin=viz_share_link)

### Выводы по разделу

В период с 1990 по 2022 гг:  

- число государственных больничных организаций снизилось на 40,5%;
- в 2.76 раза увеличилось число частных больниц;
- количество участковых (сельских) больниц уменьшилось в 229 раза.

## Дома ребенка. Ресурсы и деятельность организаций здравоохранения.

### Предположение

За период 1990 - 2022 количество детей в домах ребенка снизилось в 3 раза.

### Загрузка и предобработка файлов

In [ ]:
сhildren_homes_2005 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2005/Ресурсы и деятельность учреждений здравоохранения/дома ребенка.csv')

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2005.columns = ['name', '1990', '1995', '2000', '2001', '2002', '2003', '2004']

сhildren_homes_2005 = сhildren_homes_2005.drop(index = 0).reset_index(drop=True)

In [ ]:
сhildren_homes_2005.info()

In [ ]:
сhildren_homes_2005

In [ ]:
сhildren_homes_2007 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2007/Ресурсы и деятельность учреждений здравоохранения/дома ребенка.csv')

сhildren_homes_2007

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2007.columns = ['name', '1990', '1995', '2000', '2005', '2006']

сhildren_homes_2007 = сhildren_homes_2007.drop(index = 0).reset_index(drop=True)

In [ ]:
сhildren_homes_2007.head()

In [ ]:
сhildren_homes_2009 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2009/Ресурсы и деятельность учреждений здравоохранения/дома ребенка.csv')

сhildren_homes_2009

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2009.columns = ['name', '1995', '2000', '2005', '2007', '2008']

сhildren_homes_2009 = сhildren_homes_2009.drop(index = 0).reset_index(drop=True)

In [ ]:
сhildren_homes_2009.head()

In [ ]:
сhildren_homes_2011 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2011/Ресурсы и деятельность учреждений здравоохранения/дома ребенка.csv')

сhildren_homes_2011

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2011.columns = ['name', '1995', '2000', '2005', '2009', '2010']

сhildren_homes_2011 = сhildren_homes_2011.drop(index = 0).reset_index(drop=True)

In [ ]:
сhildren_homes_2011

In [ ]:
сhildren_homes_2013 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2013/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.4. дома ребенка1).csv')

сhildren_homes_2013

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2013.columns = ['name', '2000', '2005', '2010', '2011', '2012']

сhildren_homes_2013

In [ ]:
сhildren_homes_2015 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2015/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.4. дома ребенка1).csv')

сhildren_homes_2015

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2015.columns = ['name', '2005', '2010', '2011', '2012', '2013', '2014']

сhildren_homes_2015.head()

In [ ]:
сhildren_homes_2017 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2017/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.4. дома ребенка1).csv')

сhildren_homes_2017

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2017.columns = ['name', '2005', '2010', '2013', '2014', '2015', '2016']

сhildren_homes_2017.head()

In [ ]:
сhildren_homes_2019 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2019/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.4. дома ребенка1).csv')

сhildren_homes_2019

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2019.columns = ['name', '2005', '2010', '2015', '2016', '2017', '2018']

сhildren_homes_2019.head()

In [ ]:
сhildren_homes_2021 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2021/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.4. дома ребенка1).csv')

сhildren_homes_2021

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2021.columns = ['name', '2005', '2010', '2015', '2018', '2019', '2020']

сhildren_homes_2021.head()

In [ ]:
сhildren_homes_2023 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2023/3.4. ДОМА РЕБЕНКА1) - 3.4.csv')

сhildren_homes_2023

Сделаем понятными названия столбцов - по годам

In [ ]:
сhildren_homes_2023.columns = ['name', '2010', '2015', '2019', '2020', '2021', '2022']

сhildren_homes_2023 = сhildren_homes_2023.drop(index = 0).reset_index(drop=True)

In [ ]:
сhildren_homes_2023.head()

Для унификации названий в колонке name, возьмем за основу названия 2005 года, сохраним их в отдельную переменную и запишем по остальным годам эти названия в колонку name

In [ ]:
name_unified = сhildren_homes_2005['name']

In [ ]:
сhildren_homes_2007['name'] = name_unified

In [ ]:
сhildren_homes_2009['name'] = name_unified

In [ ]:
сhildren_homes_2011['name'] = name_unified

In [ ]:
сhildren_homes_2013['name'] = name_unified

In [ ]:
сhildren_homes_2015['name'] = name_unified

In [ ]:
сhildren_homes_2017['name'] = name_unified

In [ ]:
сhildren_homes_2019['name'] = name_unified

In [ ]:
сhildren_homes_2021['name'] = name_unified

In [ ]:
сhildren_homes_2023['name'] = name_unified

### Соединяем данные из всех папок в один датафрейм

In [ ]:
сhildren_homes_1 = сhildren_homes_2005.merge(сhildren_homes_2007, how='left')

сhildren_homes_1

In [ ]:
сhildren_homes_2 = сhildren_homes_1.merge(сhildren_homes_2009, how='left')

сhildren_homes_2

In [ ]:
сhildren_homes_3 = сhildren_homes_2.merge(сhildren_homes_2011, how='left')

сhildren_homes_3

In [ ]:
сhildren_homes_4 = сhildren_homes_3.merge(сhildren_homes_2013, how='left')

сhildren_homes_4

In [ ]:
сhildren_homes_5 = сhildren_homes_4.merge(сhildren_homes_2015, how='left')

сhildren_homes_5

In [ ]:
сhildren_homes_6 = сhildren_homes_5.merge(сhildren_homes_2017, how='left')

сhildren_homes_6

In [ ]:
сhildren_homes_7 = сhildren_homes_6.merge(сhildren_homes_2019, how='left')

сhildren_homes_7

In [ ]:
сhildren_homes_7.info()

Заменим неразрывный пробел в датафрейме `сhildren_homes_2021` для дальнейшего преобразования этих колонок к типу float с целью соединеия в общий датафрейм.

In [ ]:
сhildren_homes_2021.iloc[:,1:] = сhildren_homes_2021.iloc[:,1:].apply(lambda x: x.str.replace('\xa0', ''))

Теперь произведём в цикле изменения всех записей в колонках по годам с типа object на тип float64

In [ ]:
for names, values in сhildren_homes_2021.iloc[:,1:].items():
    сhildren_homes_2021[names] = values.astype('float')

In [ ]:
сhildren_homes_2021.info()

In [ ]:
сhildren_homes_8 = сhildren_homes_7.merge(сhildren_homes_2021, how='left')

сhildren_homes_8

Изменим тип данных датафрейма `сhildren_homes_2023` в колонках по годам с object на тип float64 в цикле. Но перед этим удалим неразрывный пробел.

In [ ]:
сhildren_homes_2023.iloc[:, 1:] = сhildren_homes_2023.iloc[:, 1:].apply(lambda x: x.str.replace('\xa0', ''))

In [ ]:
for names, values in сhildren_homes_2023.iloc[:,1:].items():
    сhildren_homes_2023[names] = values.astype('float')

In [ ]:
сhildren_homes_total = сhildren_homes_8.merge(сhildren_homes_2023, how='left')

сhildren_homes_total

Удалим образовавшиеся дубликаты

In [ ]:
сhildren_homes_total = сhildren_homes_total.drop_duplicates().reset_index(drop=True)

сhildren_homes_total

Перед сохранением датафрейма изменим все значения float на int. Для этого сначала все nan заменим на "0"

In [ ]:
сhildren_homes_total = сhildren_homes_total.fillna(0)

In [ ]:
for names, values in сhildren_homes_total.iloc[:, 1:].items():
    сhildren_homes_total[names] = values.astype('int')

Перед сохранением таблицы, сначала её транспонируем

In [ ]:
сhildren_homes_total = сhildren_homes_total.set_index('name')

сhildren_homes_total = сhildren_homes_total.T

Также приведём все названия к нижнему регистру

In [ ]:
сhildren_homes_total.columns = [x.lower() for x in сhildren_homes_total.columns]

In [ ]:
сhildren_homes_total

Сохраняем датасет в файл

In [ ]:
сhildren_homes_total.to_csv('сhildren_homes_1990_2022.csv')

### Ссылка на дашборд

Дашборд выполнен в программе Tableau и доступен по [ссылке](https://public.tableau.com/views/hildren_homes_1990_2022/Dashboard1?:language=en-US&:sid=&:display_count=n&:origin=viz_share_link)

### Выводы по разделу

В период с 1990 по 2022 гг количество детей в домах ребенка снизилось в 3,5 раза

## Коечный фонд по специализации в Субъектах Российской Федерации. Ресурсы и деятельность организаций здравоохранения.

### Предположение

За период 2004 - 2012 в структуре коечного фонда количество онкологических и инфекционных коек выросло, а количество коек для беременных, рожениц и родильниц снизилось на 30%. Но при этом количество амбулаторных медицинских организаций (поликлиник) выросло более чем на 30%.

### Загрузка данных

In [ ]:
bed_fund_2005 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2005/Ресурсы и деятельность учреждений здравоохранения/число больничных коек по специализации по регионам российской федерации в 2004 г..csv')

bed_fund_2005

In [ ]:
bed_fund_2007 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2007/Ресурсы и деятельность учреждений здравоохранения/число больничных коек по специализации по субъектам российской федерации в 2006 г..csv')

bed_fund_2007

In [ ]:
bed_fund_2009 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2009/Ресурсы и деятельность учреждений здравоохранения/число больничных коек по специализации по субъектам российской федерации в 2008 г. начало.csv')

bed_fund_2009

In [ ]:
bed_fund_2011 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2011/Ресурсы и деятельность учреждений здравоохранения/число больничных коек по специализации по субъектам российской федерации в 2010 г. начало.csv')

bed_fund_2011

In [ ]:
bed_fund_2013 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2013/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.15. число больничных коек по специализации по субъектам российской федерации в 2012 г..csv')

bed_fund_2013

In [ ]:
bed_fund_2015 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2015/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.2. число больничных коек по специализации.csv')

bed_fund_2015

In [ ]:
bed_fund_2017 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2017/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.2. число больничных коек по специализации.csv')

bed_fund_2017

In [ ]:
bed_fund_2019 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2019/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.2. число больничных коек по специализации.csv')

bed_fund_2019

In [ ]:
bed_fund_2021 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2021/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.2. число больничных коек по специализации.csv')

bed_fund_2021

In [ ]:
bed_fund_2023 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2023/3.2. ЧИСЛО БОЛЬНИЧНЫХ КОЕК ПО СПЕЦИАЛИЗАЦИИ - 3.2.csv')

bed_fund_2023

### Предобработка данных

#### Данные из папки за 2005 год

Возьмем первую строку датафрейма и сделаем название колонок ею и потом сразу удалим её

In [ ]:
bed_fund_2005.columns = list(bed_fund_2005.iloc[0])

In [ ]:
bed_fund_2005 = bed_fund_2005.drop(index = 0)

Теперь удалим все тире в названиях колонок

In [ ]:
bed_fund_2005.columns = bed_fund_2005.columns.fillna('субъект')

In [ ]:
bed_fund_2005.columns = [x.replace('-', '') for x in bed_fund_2005.columns]

Также удалим все пробелы между словами, сохраним их в отдельную переменную и передадим ей новый список названий колонок и запишем эти новые названия колонок в датафрейм

In [ ]:
column = list(bed_fund_2005.columns)

new_column = [i.replace(' ', '') for i in column]

bed_fund_2005.columns = new_column

Добавим самым первым столбцов год, из которого мы берём данные для дальнейшей визуализации в динамике

In [ ]:
bed_fund_2005.insert(0, 'год', '2004')

In [ ]:
bed_fund_2005.head()

#### Данные из папки 2007 года

In [ ]:
bed_fund_2007.head()

Для датафрейма `bed_fund_2007` проведем аналогичные операции по данным из папки за 2005 год

In [ ]:
bed_fund_2007.columns = list(bed_fund_2007.iloc[0])

In [ ]:
bed_fund_2007 = bed_fund_2007.drop(index = 0)

In [ ]:
bed_fund_2007.columns = bed_fund_2007.columns.fillna('субъект')

In [ ]:
bed_fund_2007.columns = [x.replace('-', '') for x in bed_fund_2007.columns]

In [ ]:
column = list(bed_fund_2007.columns)

new_column = [i.replace(' ', '') for i in column]

bed_fund_2007.columns = new_column

In [ ]:
bed_fund_2007.insert(0, 'год', '2006')

bed_fund_2007.head()

#### Данные из папки 2009 года

In [ ]:
bed_fund_2009.head()

In [ ]:
bed_fund_2009.columns = list(bed_fund_2009.iloc[0])

bed_fund_2009 = bed_fund_2009.drop(index = 0)

bed_fund_2009.columns = bed_fund_2009.columns.fillna('субъект')

bed_fund_2009.columns = [x.replace('-', '') for x in bed_fund_2009.columns]

column = list(bed_fund_2009.columns)

new_column = [i.replace(' ', '') for i in column]

bed_fund_2009.columns = new_column

In [ ]:
bed_fund_2009.insert(0, 'год', '2008')

bed_fund_2009.head()

#### Данные из папки за 2011 год

In [ ]:
bed_fund_2011

In [ ]:
bed_fund_2011.columns = list(bed_fund_2011.iloc[0])

bed_fund_2011 = bed_fund_2011.drop(index = 0)

bed_fund_2011.columns = bed_fund_2011.columns.fillna('субъект')

bed_fund_2011.columns = [x.replace('-', '') for x in bed_fund_2011.columns]

column = list(bed_fund_2011.columns)

new_column = [i.replace(' ', '') for i in column]

bed_fund_2011.columns = new_column

In [ ]:
bed_fund_2011.insert(0, 'год', '2010')

bed_fund_2011.head()

#### Данные из папки за 2013 год

In [ ]:
bed_fund_2013.head()

Для начала удалим символ "\n" в названиях колонок датафрейма

In [ ]:
bed_fund_2013.columns = [x.replace('\n',' ') for x in bed_fund_2013.columns]

Теперь удалим все тире между названиями колонок

In [ ]:
bed_fund_2013.columns = [x.replace('-', '') for x in bed_fund_2013.columns]

Теперь удалим все пробелы в названиях колонок

In [ ]:
column = list(bed_fund_2013.columns)

new_column = [i.replace(' ', '') for i in column]

bed_fund_2013.columns = new_column

In [ ]:
bed_fund_2013 = bed_fund_2013.rename(columns={'Unnamed:0':'субъект'})

bed_fund_2013.head()

Теперь удалим символ '\n' во всем датафрейме 

In [ ]:
bed_fund_2013 = bed_fund_2013.replace(r'\n', '', regex=True)

In [ ]:
bed_fund_2013.insert(0, 'год', '2012')

bed_fund_2013.head()

### Сохранение данных

In [ ]:
bed_fund_2005.to_csv('bed_fund_2005.csv')

In [ ]:
bed_fund_2007.to_csv('bed_fund_2007.csv')

In [ ]:
bed_fund_2009.to_csv('bed_fund_2009.csv')

In [ ]:
bed_fund_2011.to_csv('bed_fund_2011.csv')

In [ ]:
bed_fund_2013.to_csv('bed_fund_2013.csv')

### Дополнительное соединение и сохранение датасета

Соединение всеx таблиц "снизу"

In [ ]:
s = pd.concat([bed_fund_2005, bed_fund_2007], axis=0)

In [ ]:
s_1 = pd.concat([s, bed_fund_2009], axis=0)

In [ ]:
s_2 = pd.concat([s_1, bed_fund_2011], axis=0)

In [ ]:
s_3 = pd.concat([s_2, bed_fund_2013], axis=0)

Сохраним итоговую таблицы в датасет с названием test_3

In [ ]:
s_3.to_csv('test_3.csv')

### Ссылка на дашборд

Дашборд выполнен в программе Tableau и доступен по [ссылке](https://public.tableau.com/views/bed_fund_corrected/sheet4?:language=en-US&:sid=&:display_count=n&:origin=viz_share_link)

### Выводы по разделу

За период 2004 - 2012:
- кол-во инфекционных коек:
    - по федеральным округам кол-во коек уменьшилось в 1,9 раза;
    - по областям кол-во коек уменьшилось в 2,9 раза;
    - по республикам кол-во коек уменьшилось в 3 раза;
    - по 2 городам федерального значения (Москва и СПб) кол-во коек уменьшилось в 1,06 раза;
    - по автономным округам кол-во коек уменьшилось в 1,2 раза.
- кол-во онкологических коек:
    - по федеральным округам кол-во коек уменьшилось в 1,3 раза;
    - по областям кол-во коек уменьшилось в 1,9 раза;
    - по республикам кол-во коек уменьшилось в 3,1 раза;
    - по 2 городам федерального значения (Москва и СПб) кол-во коек **увеличилось** в 1,2 раза;
    - по автономным округам кол-во коек **увеличилось** в 1.08 раза.
- кол-во коек для беременных, рожениц и родильниц:
    - по федеральным округам кол-во коек уменьшилось в 1.7 раза;
    - по областям кол-во коек уменьшилось в 2,5 раза;
    - по республикам кол-во коек уменьшилось в 2,4 раза;
    - по 2 городам федерального значения (Москва и СПб) кол-во коек уменьшилось в 1.022 раза;
    - по автономным округам данных недостаточно.
    
При этом количество амбулаторных медицинских организаций (поликлиник) уменьшилось на 6%.

## Техническое состояние и благоустройство зданий лечебно-профилактических учреждений здравоохранения. Ресурсы и деятельность организаций здравоохранения.

### Предположение

За период 1990 - 2022 какова динамика снижения процентного соотношения зданий больниц в которых отсутствует водопровод, горячее водоснабжение, центральное отопление, канализация, телефонная связь?

### Загрузка данных

In [ ]:
technical_condition_2005 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2005/Ресурсы и деятельность учреждений здравоохранения/техническое состояние и благоустройство зданий лечебно-профилактических учреждений здравоохранения.csv')

technical_condition_2005

In [ ]:
technical_condition_2007 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2007/Ресурсы и деятельность учреждений здравоохранения/техническое состояние и благоустройство зданий лечебно - профилактических учреждений здравоохранения.csv')

technical_condition_2007

In [ ]:
technical_condition_2009 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2009/Ресурсы и деятельность учреждений здравоохранения/техническое состояние и благоустройство зданий лечебно- профилактических учреждений здравоохранения.csv')

technical_condition_2009

In [ ]:
technical_condition_2011 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2011/Ресурсы и деятельность учреждений здравоохранения/техническое состояние и благоустройство зданий лечебно-профилактических учреждений здравоохранения.csv')

technical_condition_2011

In [ ]:
technical_condition_2013 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2013/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.10. техническое состояние и благоустройство зданий  лечебно-профилактических организаций здравоохранения1).csv')

technical_condition_2013

In [ ]:
technical_condition_2015 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2015/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.10. техническое состояние и благоустройство зданий  лечебно-профилактических организаций здравоохранения1).csv')

technical_condition_2015

In [ ]:
technical_condition_2017 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2017/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.12. техническое состояние и благоустройство зданий  лечебно-профилактических организаций здравоохранения1).csv')

technical_condition_2017

In [ ]:
technical_condition_2019 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2019/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.12. техническое состояние и благоустройство зданий  лечебно-профилактических организаций здравоохранения1).csv')

technical_condition_2019

In [ ]:
technical_condition_2021 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2021/РЕСУРСЫ И ДЕЯТЕЛЬНОСТЬ ОРГАНИЗАЦИЙ /3.12. техническое состояние и благоустройство зданий  медицинских организаций здравоохранения1).csv')

technical_condition_2021

In [ ]:
technical_condition_2023 = pd.read_csv('/Users/mac/Desktop/проекты в Яндекс Практикуме/Датасеты и проекты/Мастерская/данные/2023/3.12. ТЕХНИЧЕСКОЕ СОСТОЯНИЕ И БЛАГОУСТРОЙСТВО ЗДАНИЙ МЕДИЦИНСКИХ ОРГАНИЗАЦИЙ ЗДРАВООХРАНЕНИЯ1) - 3.12.csv')

technical_condition_2023

### Предобработка данных

Возьмем первую строку датафрейма и сделаем из неё название колонок и потом сразу удалим её. Для этого напишем функцию и передадим ей датафреймы по очереди вплоть до 2011 года и для 2023 года тоже.

In [ ]:
def column_new(data):
    data.columns = list(data.iloc[0])
    data = data.drop(index = 0).reset_index(drop=True)
    data.rename(columns = {np.nan:'name'}, inplace=True)
    return data

In [ ]:
technical_condition_2005 = column_new(technical_condition_2005)

In [ ]:
technical_condition_2007 = column_new(technical_condition_2007)

In [ ]:
technical_condition_2009 = column_new(technical_condition_2009)

In [ ]:
technical_condition_2011 = column_new(technical_condition_2011)

In [ ]:
technical_condition_2023 = column_new(technical_condition_2023)

С датафреймами из папок с 2013 по 2021 изменим названия 1 колонки с `Unnamed: 0	` на `name` с целью дальнейшего корректного соединения.

In [ ]:
technical_condition_2013.rename(columns={'Unnamed: 0':'name'}, inplace=True)

technical_condition_2015.rename(columns={'Unnamed: 0':'name'}, inplace=True)

technical_condition_2017.rename(columns={'Unnamed: 0':'name'}, inplace=True)

technical_condition_2019.rename(columns={'Unnamed: 0':'name'}, inplace=True)

technical_condition_2021.rename(columns={'Unnamed: 0':'name'}, inplace=True)

Создадим список датафреймов и проверим их длину

In [ ]:
t = [technical_condition_2007, 
     technical_condition_2009,
    technical_condition_2011,
    technical_condition_2013,
    technical_condition_2015,
    technical_condition_2017,
    technical_condition_2019,
    technical_condition_2021,
    technical_condition_2023]

In [ ]:
for el in t:
    print(len(el))

Все получившиеся датафреймы имеют одинаковую длину и для унификации строк с индексами 0 и 12 создадим список и передадим его каждому датафрейму. За образец возьмём названия этих строк по индексам за 2005-2009 года, так как они кажутся наиболее полными.

Напишем функцию, которая будет прпнимать датафрейм и шаблонный датафрейм (для названия колонок) из папки 2005 года и вернёт изменённые строки с индексами 0 и 12 на унифицированные. Также в этой функции предусмотрим изменение название колонки `name` во всех датасетах на наименования из колонки `name` шаблонного датафрейма (опять же с целью унификации названий и дальнейшего соединения).

In [ ]:
technical_condition_2009.columns

In [ ]:
def row_new(data, data_pattern):
    mask_row_0 = data_pattern['1995'].values[0]
    
    mask_row_12 = data_pattern['1995'].values[12]
    
    data.iloc[0,1:] = mask_row_0
    
    data.iloc[12,1:] = mask_row_12
    
    mask_column = data_pattern['name']
    
    data['name'] = mask_column
    
    return data

Теперь вновь написанной функции передадим все данные из папок с 2005 по 2023

In [ ]:
с = row_new(technical_condition_2007, technical_condition_2005)

technical_condition_2009 = row_new(technical_condition_2009, technical_condition_2005)

technical_condition_2011 = row_new(technical_condition_2011, technical_condition_2005)

technical_condition_2013 = row_new(technical_condition_2013, technical_condition_2005)

technical_condition_2015 = row_new(technical_condition_2015, technical_condition_2005)

technical_condition_2017 = row_new(technical_condition_2017, technical_condition_2005)

technical_condition_2019 = row_new(technical_condition_2019, technical_condition_2005)

technical_condition_2021 = row_new(technical_condition_2021, technical_condition_2005)

technical_condition_2023 = row_new(technical_condition_2023, technical_condition_2005)

### Соединение в единый датасет

Соединими все таблицы в единый датасет

In [ ]:
technical_condition_1 = pd.concat([technical_condition_2005, technical_condition_2007], axis=1)

In [ ]:
technical_condition_2 = pd.concat([technical_condition_1, technical_condition_2009], axis=1)

In [ ]:
technical_condition_3 = pd.concat([technical_condition_2, technical_condition_2011], axis=1)

In [ ]:
technical_condition_4 = pd.concat([technical_condition_3, technical_condition_2013], axis=1)

In [ ]:
technical_condition_5 = pd.concat([technical_condition_4, technical_condition_2015], axis=1)

In [ ]:
technical_condition_6 = pd.concat([technical_condition_5, technical_condition_2017], axis=1)

In [ ]:
technical_condition_7 = pd.concat([technical_condition_6, technical_condition_2019], axis=1)

In [ ]:
technical_condition_8 = pd.concat([technical_condition_7, technical_condition_2021], axis=1)

In [ ]:
technical_condition_total = pd.concat([technical_condition_8, technical_condition_2023], axis=1)

Теперь удалим дублирующиеся столбцы по наименванию года

In [ ]:
technical_condition_total = technical_condition_total.loc[:,~technical_condition_total.columns.duplicated()].copy()

technical_condition_total

### Сохранение получившихся датасетов и разделение их на стационар и амбулатории

Из исходного датафрейма сохраним стационарные учреждения в один датасет, а амбулаторные учреждения в другой датасет

Стационарные учреждения - technical_condition_hospital

In [ ]:
technical_condition_hospital = technical_condition_total.iloc[:12,:].copy()

technical_condition_hospital.drop(index=0, inplace=True)

technical_condition_hospital

Амбулаторные учреждения - technical_condition_outpatient_clinic

In [ ]:
technical_condition_outpatient_clinic = technical_condition_total.iloc[12:,:].copy()

technical_condition_outpatient_clinic.reset_index(drop=True, inplace=True)

technical_condition_outpatient_clinic.drop(index=0, inplace=True)

In [ ]:
technical_condition_outpatient_clinic

Для корректной работы с данными при визуализации, изменим все символы запятых на точки. Основная цель этих изменений - запятые между цифрами.

In [ ]:
technical_condition_hospital = technical_condition_hospital.apply(lambda x: x.str.replace(',', '.'))

technical_condition_outpatient_clinic = technical_condition_outpatient_clinic.apply(lambda x: 
                                                                                    x.str.replace(',', '.'))

Транспонируем датасеты и сохраняем для дальнейшей их визуализации и построение дашбордов.

In [ ]:
technical_condition_hospital = technical_condition_hospital.set_index('name').T

In [ ]:
technical_condition_outpatient_clinic = technical_condition_outpatient_clinic.set_index('name').T

In [ ]:
technical_condition_hospital.to_csv('technical_condition_hospital.csv')

In [ ]:
technical_condition_outpatient_clinic.to_csv('technical_condition_outpatient_clinic.csv')

### Ссылка на дашборд

Дашборд выполнен в программе Tableau и доступен по [ссылке](https://public.tableau.com/views/technical_condition_hospital/sheet13?:language=en-US&:sid=&:display_count=n&:origin=viz_share_link)

### Вывод по разделу

Стационарные медицинские учреждения и динамика % соотношения:

- % отсутствия горячего водоснабжения снизился на 23,6%;
- % отсутствия канализации снизился на 14,6%;
- % отсутствия центрального отопления снизился на 5,8%;
- % отсутствия водопровода снизился на 10,4%;
- % отсутствия телефонной связи снизился на 5,4%.  

Амбулаторные медицинские учреждения и динамика % соотношения:

- % отсутствия горячего водоснабжения снизился на 19,8%;
- % отсутствия канализации снизился на 16,7%;
- % отсутствия центрального отопления снизился на 2,1%;
- % отсутствия водопровода снизился на 12%;
- % отсутствия телефонной связи снизился на 4%. 